### DISCUSSION TITLE
HTTP GET binary handle does not return a h.read() as described in the documentation

### DISCUSSION THREAD

**Discussion by EmmaKnijn** [cc-tweaked/CC-Tweaked#1175] [Q&A] [answered]
### Minecraft Version

1.18.x

### Version

1.100.10

### Details

I was porting my program from old Computronics to ComputerCraft:Tweaked 1.100's dfpwm playing system. I am using this as the code: https://github.com/knijn/musicify/blob/01f272e6df3dafb32aac1f13ddfffc9af4337b97/musicify.lua

Specific function that is relevant is play(), it should download the file (with the options `local h = http.get({["url"] = songID.file, ["binary"] = true, ["redirect"] = true}`)
) and shove it into the handle. I'm trying to use h.read() to read through the dfpwm file but its returning nil for the chunk, causing bogus audio to be played
![image](https://user-images.githubusercontent.com/64010592/193454636-9f09ad1e-c007-4a5c-8c62-69e1b4b2c7b9.png)

**Comment by EmmaKnijn** [comment]
I was following the guide under https://tweaked.cc/guide/speaker_audio.html Storing Audio and at the bottom it suggests https://tweaked.cc/module/fs.html#ty:BinaryReadHandle:read. As it notes at https://tweaked.cc/module/http.html#ty:Response, the binary handle returns the same methods as https://tweaked.cc/module/fs.html#ty:BinaryReadHandle

**Comment by SquidDev** [comment, marked as answer]
This is because you've already read the whole file on line 113. At this point you're at the end of the file, so there's nothing else to read!

I'd drop the `readAll` and the `for` loop, instead do a basic `while` loop - something like:

```lua
local h = http.get({["url"] = songID.file, ["binary"] = true, ["redirect"] = true})
while true do
  local chunk = h.read(16 * 1024)
  if not chunk then break end

  -- process chunk as before
end
```
